lib/fern/api.rb

```diff
@@ -8,14 +8,43 @@ module Api
 
     included do
       class_eval { class_attribute :fern }
-      self.fern = {}
-      def present(obj)
-        render json: obj
+
+      def form
+        if @form.nil?
+          p = params
+          if fern[action_name.to_sym][:form].has_key?(:key) && fern[action_name.to_sym][:form][:key]
+            p = params[fern[action_name.to_sym][:form][:key]]
+          end
+          p = p&.to_hash&.deep_symbolize_keys
+          @form ||= fern[action_name.to_sym][:form][:klass].new(p)
+        end
+        @form
+      end
+
+      def present(object, **opts)
+        render json: render_presenter(object, **opts)
+      end
+
+      def render_presenter(object, **opts)
```

Comment:
I'm a little torn about this method. It's certainly easy to grok (although we may consider drying up `fern[action_name.to_sym][:presenter]` which appears in here four times), but lines 31 and 33, and 37 and 39 are so similar, I can't help but think there's a more concise way to express the logic. Any ideas?

Reply:
```         
 if opts.any?
            fern[action_name.to_sym][:presenter].new(object).as_json(**opts)
          else
            fern[action_name.to_sym][:presenter].new(object).as_json()
          end
```

is it possible to simply just pass `**opts` in any case? I see this around much of the presenter logic

Reply:
```      
      def render_presenter(object, **opts)
        presenter = fern[action_name.to_sym][:presenter]

        if object.respond_to?(:map)
          object.map { |el| presenter.new(el).as_json(**opts) }
        else
          presenter.new(object).as_json(**opts)
        end
      end
```

Was the original intent to not pass around empty hashes if not necessary?

Reply:
Hmm. I wish I could remember! I suspect I was trying to prevent you from always having to declare your `as_json` method as taking an argument. Otherwise you'd get a `ArgumentError: wrong number of arguments (given 1, expected 0)`.